core/src/test/java/org/elasticsearch/search/aggregations/InternalAggregationTestCase.java

```diff
@@ -122,4 +138,56 @@ protected final T createUnmappedInstance(String name) {
     protected NamedWriteableRegistry getNamedWriteableRegistry() {
         return namedWriteableRegistry;
     }
+
+    protected ContextParser<Object, Aggregation> instanceParser() {
+        throw new UnsupportedOperationException("Not implemented yet");
+    }
+
+    public void testFromXContent() throws IOException {
+        try {
+            final T aggregation = createTestInstance();
+            NamedXContentRegistry xContentRegistry = new NamedXContentRegistry(
+                    singletonList(new NamedXContentRegistry.Entry(Aggregation.class, new ParseField(aggregation.getType()), instanceParser()))
+            );
```

Comment:
Oh I see. I didn't think about this method as I wanted to mimic the already existing `instanceReader()` one, but it makes sense.

I'll update the PR.